bigquery_etl/shredder/bug1751979.py

```diff
@@ -0,0 +1,251 @@
+          ) AS key,
+          value
+        )
+      )
+    FROM
+      base
+  )
+);
+
+CREATE TEMP FUNCTION sanitize_scalar(input ANY TYPE) AS (
+  (
+    WITH base AS (
+      SELECT
+        key,
+        value,
+        REGEXP_EXTRACT(key, "([^:]+:[^:]+:).*") AS prefix,
+        REGEXP_EXTRACT(key, "[^:]+:[^:]+:(.*)") AS code,
+      FROM
+        UNNEST(input)
+    )
+    SELECT
+      ARRAY_AGG(
+        STRUCT(
+          IF(
+            prefix IS NULL
+            OR code IN (
+              "none",
+              "other",
+              "hz",
+              "h_",
+              "MOZ2",
+              "MOZ4",
+              "MOZ5",
+              "MOZA",
+              "MOZB",
+              "MOZD",
+              "MOZE",
+              "MOZI",
+              "MOZM",
+              "MOZO",
+              "MOZT",
+              "MOZW",
+              "MOZSL01",
+              "MOZSL02",
+              "MOZSL03",
+              "firefox-a",
+              "firefox-b",
+              "firefox-b-1",
+              "firefox-b-ab",
+              "firefox-b-1-ab",
+              "firefox-b-d",
+              "firefox-b-1-d",
+              "firefox-b-e",
+              "firefox-b-1-e",
+              "firefox-b-m",
+              "firefox-b-1-m",
+              "firefox-b-o",
+              "firefox-b-1-o",
+              "firefox-b-lm",
+              "firefox-b-1-lm",
+              "firefox-b-lg",
+              "firefox-b-huawei-h1611",
+              "firefox-b-is-oem1",
+              "firefox-b-oem1",
+              "firefox-b-oem2",
+              "firefox-b-tinno",
+              "firefox-b-pn-wt",
+              "firefox-b-pn-wt-us",
+              "ubuntu",
+              "ffab",
+              "ffcm",
+              "ffhp",
+              "ffip",
+              "ffit",
+              "ffnt",
+              "ffocus",
+              "ffos",
+              "ffsb",
+              "fpas",
+              "fpsa",
+              "ftas",
+              "ftsa",
+              "newext",
+              "monline_dg",
+              "monline_3_dg",
+              "monline_4_dg",
+              "monline_7_dg"
+            ),
+            key,
+            CONCAT(prefix, "other.scrubbed")
+          ) AS key,
+          value
+        )
+      )
+    FROM
+      base
+  )
+);
+"""
+
+BUG_1751979_MAIN_V4_REPLACE_CLAUSE = """
+REPLACE (
+(
+  SELECT AS STRUCT
+    payload.* REPLACE (
+      (
+        SELECT AS STRUCT
+          payload.keyed_histograms.* REPLACE (
+            sanitize_search_counts(payload.keyed_histograms.search_counts) AS search_counts
+          )
+      ) AS keyed_histograms,
+      (
+        SELECT AS STRUCT
+          payload.processes.* REPLACE (
+            (
+              SELECT AS STRUCT
+                payload.processes.parent.* REPLACE (
+                  (
+                    SELECT AS STRUCT
+                      payload.processes.parent.keyed_scalars.* REPLACE (
```

Comment:
this is going to cause `payload.keyed_histograms` and `payload.processes.parent.keyed_scalars` to be empty structs instead of `NULL` if they were previously `NULL`. I'm not sure if that's a problem here, or if that's fine.

Reply:
I would expect it's rare to have these either of those fields be null. And I can't imagine a use case where the semantics would matter if we do replace it. But it's a good point. Might be worth trying to fix.

Reply:
Looks like it's in the low single-digit percents:

```
SELECT 
    COUNTIF( payload.keyed_histograms.search_counts IS NULL) / COUNT(*),
    COUNTIF( payload.processes.parent.keyed_scalars.browser_search_content_urlbar IS NULL) / COUNT(*)
  FROM
    `moz-fx-data-shared-prod.telemetry.main_1pct` AS main
  WHERE
    DATE(submission_timestamp) = '2022-02-08'
```

yields 0.015 and 0.03. So it happens.

Reply:
cool, sounds fine to ignore then.